src/MarianInterface.cpp

```diff
@@ -104,34 +97,48 @@ MarianInterface::MarianInterface(QObject *parent)
             emit pendingChanged(true);
 
             try {
-                if (model) {
-                    // Unload marian first (so we can delete loggers after that)
+                if (modelChange) {
+                    // Reconstruct the service because cpu_threads might have changed.
+                    // @TODO: don't recreate Service if cpu_threads didn't change?
+                    marian::bergamot::AsyncService::Config serviceConfig;
+                    serviceConfig.numWorkers = modelChange->settings.cpu_threads;
+                    
+                    // Free up old service first (see https://github.com/browsermt/bergamot-translator/issues/290)
                     service.reset();
 
-                    // We need to manually destroy the loggers, as marian doesn't do
-                    // that but will complain when a new marian::Config tries to 
-                    // initialise loggers with the same name.
-                    spdlog::drop("general");
-                    spdlog::drop("valid");
+                    service = std::make_unique<marian::bergamot::AsyncService>(serviceConfig);
 
-                    service.reset(new marian::bergamot::Service(MakeOptions(model->config_file, model->settings)));
+                    // Initialise a new model. Old model will be released if
+                    // service is done with it, which it is since all translation
+                    // requests are effectively blocking in this thread.
+                    auto modelConfig = makeOptions(modelChange->config_file, modelChange->settings);
+                    model = std::make_shared<marian::bergamot::TranslationModel>(modelConfig, marian::bergamot::MemoryBundle{}, modelChange->settings.cpu_threads);
```

Comment:
The WASM optimized load if I remember correctly is via `MemoryBundle` which appears to be empty here and FS loads are happening. FS loads have been slow (not just WASM). 

@XapaJIaMnu #76.

Reply:
https://github.com/jerinphilip/lemonade/blob/b8f4264084127e3da2b07ef280de8fc9fc5db725/lemonade/lib/translator.cpp#L28-L36, although I'm still not particularly happy with the load times after either. But should be faster than repeated on-disk loads.

Reply:
Thanks! To me the difference seems to be reading once, and reading once for every worker thread. I'll implement it that way, see how well it works. I assume it just reads the file into memory, and each worker still has to do the parsing on their own, but changing that (having all workers share those resources?) seems … nontrivial.